### PR TITLE
Westeros update and westerossink used for video

### PR DESCRIPTION
**Pull request by aazamansari**
1. Adding westerossink support for video playback.
2. Adding output mode handling for westeros viewbackend.
3. Added code to set position and size in case of HOLE_PUNCH_GSTREAMER

### CODE REVIEW

**Comment by albertd** [comment]
@zdobersek and @philn I think it's wise that you have a quick look at this pull

File: Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp [hunk 1]
```diff
@@ -2156,7 +2156,15 @@ void MediaPlayerPrivateGStreamer::createGSTPlayBin()
     if (videoSinkPad)
         g_signal_connect_swapped(videoSinkPad.get(), "notify::caps", G_CALLBACK(videoSinkCapsChangedCallback), this);
 #endif
+#if USE(WESTEROS_SINK)
```
Comment (philn) [inline Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp]: Can this block be moved to the base class? there's a createVideoSink() method there.

File: Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp [hunk 2]
```diff
@@ -319,6 +319,20 @@ MediaPlayerPrivateGStreamerBase::~MediaPlayerPrivateGStreamerBase()
...
+    GstElement* sinkElement = nullptr;
+    g_object_get(m_pipeline.get(), "video-sink", &sinkElement, nullptr);
+    if(sinkElement) {
+        INFO_MEDIA_MESSAGE("Setting video sink size and position to x:%d y:%d, width=%d, height=%d\n",m_position.x(),m_position.y(), m_size.width(),m_size.height());
+        GValue val = G_VALUE_INIT;
+        g_value_init(&val, G_TYPE_STRING);
+        char rectString[64];
+        sprintf(rectString,"%d,%d,%d,%d",m_position.x(), m_position.y(), m_size.width(),m_size.height());
+        g_value_set_static_string(&val,rectString);
+        g_object_set_property(G_OBJECT(sinkElement),"window_set",&val);
```
Comment (philn) [inline Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp]: Please don't bother with a GValue here and use g_object_set()

File: Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp [hunk 3]
```diff
@@ -837,6 +851,29 @@ void MediaPlayerPrivateGStreamerBase::setSize(const IntSize& size)
...
 }
 
+void MediaPlayerPrivateGStreamerBase::setPosition(const IntPoint& position)
+{
+    if (position == m_position) {
+        return;
+    }
+    m_position = position;
+
+    if(m_pipeline) {
```
Comment (philn) [inline Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp]: if (!m_pipeline) return;

File: Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp [hunk 4]
```diff
@@ -837,6 +851,29 @@ void MediaPlayerPrivateGStreamerBase::setSize(const IntSize& size)
...
+        GstElement* sinkElement = nullptr;
+        g_object_get(m_pipeline.get(), "video-sink", &sinkElement, nullptr);
+        if(sinkElement) {
+            INFO_MEDIA_MESSAGE("Setting video sink size and position to x:%d y:%d, width=%d, height=%d\n",m_position.x(),m_position.y(), m_size.width(),m_size.height());
+            GValue val = G_VALUE_INIT;
+            g_value_init(&val, G_TYPE_STRING);
+            char rectString[64];
+            sprintf(rectString,"%d,%d,%d,%d",m_position.x(), m_position.y(), m_size.width(),m_size.height());
+            g_value_set_static_string(&val,rectString);
+            g_object_set_property(G_OBJECT(sinkElement),"window_set",&val);
```
Comment (philn) [inline Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp]: Use g_object_set()

File: Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp (hunk 1 above)
Comment (aazamansari) [inline Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp]: We use WESTEROS_SINK when HOLE_PUNCH_GSTREAMER=ON and createVideoSink is called whtn HOLE_PUNCH_GSTREAMER=OFF. So moving it to base may not work.

File: Source/cmake/OptionsWPE.cmake [hunk 5]
```diff
@@ -69,7 +70,7 @@ find_package(ICU REQUIRED)
 find_package(Threads REQUIRED)
 find_package(ZLIB REQUIRED)
 find_package(GLIB 2.38.0 REQUIRED COMPONENTS gio gobject gthread gmodule)
-find_package(GnuTLS 3.0.0 REQUIRED)
+find_package(GnuTLS 2.12.23 REQUIRED)
```
Comment (zdobersek) [inline Source/cmake/OptionsWPE.cmake]: I'll investigate how low we can drop this, but it's OK for now.

**Comment by philn** [comment]
Merge manually.

**Pull request closed without merging**